tests/Unit/TestHelpers.hpp

```diff
@@ -91,6 +91,8 @@ T serialize_and_deserialize(const T& t) {
 }
 
 /// Test for copy semantics assuming operator== is implement correctly
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wparentheses"
```

Comment:
adding parentheses gives useless error messages when there is an error (e.g.  bool = false instead of a = bla; b = bla;)